mitiq/benchmarks/tests/test_qpe_circuits.py

```diff
@@ -0,0 +1,204 @@
+            ),
+            cirq.Moment(
+                cirq.CNOT(qubits[2], qubits[3]),
+            ),
+            cirq.Moment(
+                cirq.CNOT(qubits[1], qubits[3]),
+            ),
+            cirq.Moment(
+                cirq.CNOT(qubits[1], qubits[3]),
+            ),
+            cirq.Moment(
+                cirq.CNOT(qubits[0], qubits[3]),
+            ),
+            cirq.Moment(
+                cirq.CNOT(qubits[0], qubits[3]),
+            ),
+            cirq.Moment(
+                cirq.CNOT(qubits[0], qubits[3]),
+            ),
+            cirq.Moment(
+                cirq.CNOT(qubits[0], qubits[3]),
+            ),
+            cirq.Moment(
+                cirq.SWAP(qubits[0], qubits[2]),
+            ),
+            cirq.Moment(
+                cirq.H(qubits[0]),
+                cirq.H(qubits[1]),
+                cirq.H(qubits[2]),
+            ),
+            cirq.Moment(
+                (cirq.CNOT**-1.0).on(qubits[0], qubits[1]),
+            ),
+            cirq.Moment(
+                (cirq.CNOT**-1.0).on(qubits[0], qubits[2]),
+            ),
+            cirq.Moment(
+                (cirq.CNOT**-1.0).on(qubits[1], qubits[2]),
+            ),
+        ]
+    )
+    assert _equal(generated_circuit, expected_circuit)
+
+
+def test_circuit_QPE_TGate():
+    generated_circuit = generate_qpe_circuit(3, cirq.T)
+    qubits = cirq.LineQubit.range(3 + 1)
+    expected_circuit = cirq.Circuit(
+        [
+            cirq.Moment(
+                cirq.H(qubits[0]),
+                cirq.H(qubits[1]),
+                cirq.H(qubits[2]),
+            ),
+            cirq.Moment(
+                (cirq.CZ**0.25).on(qubits[2], qubits[3]),
+            ),
+            cirq.Moment(
+                (cirq.CZ**0.25).on(qubits[1], qubits[3]),
+            ),
+            cirq.Moment(
+                (cirq.CZ**0.25).on(qubits[1], qubits[3]),
+            ),
+            cirq.Moment(
+                (cirq.CZ**0.25).on(qubits[0], qubits[3]),
+            ),
+            cirq.Moment(
+                (cirq.CZ**0.25).on(qubits[0], qubits[3]),
+            ),
+            cirq.Moment(
+                (cirq.CZ**0.25).on(qubits[0], qubits[3]),
+            ),
+            cirq.Moment(
+                (cirq.CZ**0.25).on(qubits[0], qubits[3]),
+            ),
+            cirq.Moment(
+                cirq.SWAP(qubits[0], qubits[2]),
+            ),
+            cirq.Moment(
+                cirq.H(qubits[0]),
+                cirq.H(qubits[1]),
+                cirq.H(qubits[2]),
+            ),
+            cirq.Moment(
+                (cirq.CZ**-0.25).on(qubits[0], qubits[1]),
+            ),
+            cirq.Moment(
+                (cirq.CZ**-0.25).on(qubits[0], qubits[2]),
+            ),
+            cirq.Moment(
+                (cirq.CZ**-0.25).on(qubits[1], qubits[2]),
+            ),
+        ]
+    )
+    assert _equal(generated_circuit, expected_circuit)
+
+
+def test_phase_angle_estimation():
```

Comment:
Skipped adding a test for the T-gate because I was not sure about comparing two histogram plots.